storage/src/backend/registry.rs

```diff
@@ -282,10 +298,62 @@ struct RegistryReader {
     blob_id: String,
     connection: Arc<Connection>,
     state: Arc<RegistryState>,
+    mirror_state: Arc<MirrorState>,
     metrics: Arc<BackendMetrics>,
 }
 
+struct MirrorState {
+    mirrors: Vec<Mirror>,
+    index: ArcSwap<Option<usize>>,
+}
+
+#[derive(Debug)]
+struct Mirror {
+    state: RegistryState,
```

Comment:
How about use `Option<RegistryState>`, only set to `Some` if current mirror works fine.

Reply:
Maybe it must be wrapped in ArcSwap `ArcSwap<Option<RegistryState>>` ?

Reply:
I think it may not be necessary to modify the value of `Mirror.state` because it contains state information. If mirror is restored (not implemented yet) the state needs to be regenerated.